cg/apps/demultiplex/sample_sheet/index.py

```diff
@@ -194,7 +202,9 @@ def adapt_indexes_for_sample(
     3. Assigns the indexes to the sample attributes index1 and index2."""
     index1, index2 = get_index_pair(sample=sample)
     index_length = len(index1)
-    if is_padding_needed(index_cycles=index_cycles, sample_index_length=index_length):
+    if type(sample) == FlowCellSampleBcl2Fastq and is_padding_needed(
```

Comment:
```suggestion
    if isInstance(sample, FlowCellSampleBcl2Fastq) and is_padding_needed(
```
Would this work too? I think `type` fetches the class object

Reply:
fixed